Source/Core/Core/NetPlayClient.cpp

```diff
@@ -632,6 +636,52 @@ void NetPlayClient::SendAsync(sf::Packet&& packet)
 // called from ---NETPLAY--- thread
 void NetPlayClient::ThreadFunc()
 {
+  bool qos_success = false;
```

Comment:
This, along with the closing code below should be extracted to their own functions which are called in this function.

Reply:
Any thoughts on moving it to a class instead of just functions, so a ctor/dtor can take care of the initialization/cleanup rather than `#ifdef`-ing a bunch of platform specific stuff right in here?
If it weren't for the Windows-specific cleanup at the end, I'd probably just shove it into an initialization method (one per platform perhaps; easier to read than inline `#ifdef`)

Reply:
This works too (and is more in line with the single responsibility principle)